packages/bierzo-wallet/src/routes/registerName/index.tsx

```diff
@@ -186,6 +140,70 @@ const RegisterUsername = (): JSX.Element => {
     };
   }, [addressesSorted, bnsIdentity]);
 
+  const validate = async (values: object): Promise<object> => {
+    const formValues = values as FormValues;
+    const errors: ValidationError = {};
+
+    const username = formValues[REGISTER_USERNAME_FIELD];
+    if (!username) {
+      errors[REGISTER_USERNAME_FIELD] = "Required";
+      return errors;
+    }
+
+    const checkResult = isValidIov(username);
+
+    switch (checkResult) {
+      case "not_iov":
+        errors[REGISTER_USERNAME_FIELD] = "IOV starname must include *iov";
+        break;
+      case "wrong_number_of_asterisks":
+        errors[REGISTER_USERNAME_FIELD] = "IOV starname must include only one namespace";
+        break;
+      case "too_short":
+        errors[REGISTER_USERNAME_FIELD] = "IOV starname should be at least 3 characters";
+        break;
+      case "too_long":
+        errors[REGISTER_USERNAME_FIELD] = "IOV starname should be maximum 64 characters";
+        break;
+      case "wrong_chars":
+        errors[REGISTER_USERNAME_FIELD] =
+          "IOV starname should contain 'abcdefghijklmnopqrstuvwxyz0123456789-_.' characters only";
+        break;
+      case "valid":
+        break;
+      default:
+        throw new Error(`"Unknown IOV starname validation error: ${checkResult}`);
+    }
+
+    if (checkResult !== "valid") {
+      return errors;
+    }
+
+    const connection = await getConnectionForBns();
+    const usernames = await connection.getUsernames({ username });
+    if (usernames.length > 0) {
+      errors[REGISTER_USERNAME_FIELD] = "Personalized address already exists";
+      return errors;
+    }
+
+    const addressesToRegister = getChainAddressPairsFromValues(formValues, addressesSorted);
+    for (let index = 0; index < addressesToRegister.length; index++) {
+      const address = addressesToRegister[index];
+      const codec = await getCodecForChainId(address.chainId);
+      if (!codec.isValidAddress(address.address)) {
+        const addressFieldId = Object.entries(formValues).find(([_id, value]) => {
+          if (value === address.address) return true;
+          return false;
+        });
+        if (addressFieldId) {
+          errors[addressFieldId[0]] = "Not valid blockchain address";
```

Comment:
if `addressField` contains the field ID, shouldn't it be `errors[addressFieldId]`?

Reply:
No. `addressFieldId` contains array of `key` and `value` (see `Object.entries(formValues)`), and I need `key` because `key` === input field `name`

Reply:
I will change naming.

Reply:
Thanks for clarification